example.env

```diff
@@ -21,3 +21,5 @@ EmailSender__Smtp__Password=Password123
 
 ADMINER_PORT=1337
 
+Feature__AccountRegistration=true
```

Comment:
Doesn't the actual staging environment read from that too though? In that environment, it should be read from environment variables.

Reply:
This is how configuration is loaded in ASP (by default):

Configuration is read from multiple sources, if sources contain the same config keys, the last loaded source takes precedence.

The loading order is:

1. 'appsettings.json`
2. `appsettings.{EnvironmentName}.json`
3. Environment variables
4. Command-line arguments

So you can set the default config in `appsettings.json` and override only parts of the configuration with environment variables when needed.
 https://devblogs.microsoft.com/premier-developer/order-of-precedence-when-configuring-asp-net-core/

----

We have one customization: in the local dev scenario (not deployments), the `.env` file is read first (now that I think about it, maybe it should be last).

Reply:
I'm aware of that, and I think for simplicity, any given config value should be read in one place or the other. Rather, I wouldn't want this to break just because we decided later to reorder the configs.

Reply:
Maybe we remove `.env`'s precedence for local dev, so that it's consistent with other environments. Then we can have default values in `appsettings.json`.

Reply:
Good idea.